src/lexer.in.cc

```diff
@@ -104,12 +104,9 @@ const char* Lexer::TokenErrorHint(Token expected) {
 
 string Lexer::DescribeLastError() {
   if (last_token_) {
-    switch (last_token_[0]) {
-    case '\t':
-      return "tabs are not allowed, use spaces";
-    }
+	return "lexing error <"+string(last_token_)+">";
```

Comment:
You might want to add a static Lexer::TokenToString() method (which would implement the same code as the one you added for the test in the second commit) and use this here instead.

Reply:
this is for error (reason), not token (type), IOW valid only for Token==ERROR

Reply:
Oh, I see, thanks for the clarification then :)